quickwit/quickwit-config/src/source_config/mod.rs

```diff
@@ -346,6 +350,38 @@ pub struct VecSourceParams {
 #[serde(deny_unknown_fields)]
 pub struct VoidSourceParams;
 
+#[derive(Clone, Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize, utoipa::ToSchema)]
+#[serde(deny_unknown_fields)]
+pub struct PulsarSourceParams {
+    /// List of the topics that the source consumes.
+    pub topics: Vec<String>,
+    /// The name of the topic subscription.
+    pub subscription: String,
```

Comment:
I'm not sure what is the purpose of this `subscription` parameter. Maybe rename it `subscription_name`. Could we also have a `default_subscription_name` if it makes sense?

Reply:
Honestly this might be safer to not have this be configurable and instead generate this ourselves.

Reply:
Yes, for the Kafka source, users ended up using `quickwit` as a group ID for all their sources... so we removed it.

Reply:
I have updated this and removed the parameter.

The reason for removal is that this is potentially a foot gun for users when creating distributed indexers where we may silently fail (pulsar-rs will repeatedly try to re-connect) if two subscriptions with the same name try to consume the source.

Reply:
I've made this default to `quickwit-indexer` this is left configurable in case someone has some particular use case for it as subscription names have some semantic meaning to how consumers behave.

Reply:
That's no bueno. We want a unique subscription name per source. Use `quickwit-{index_id}-{source_id}`. Otherwise, with a unique subscription name such as `quickwit-indexer`, consumers will consume cross-indexes.

Reply:
I could be wrong because I'm mapping what I know from Kafka, and I don't know much of Pulsar, but I suspect subscription name is the primary key that determines which consumers are sharing the same set of partitions/topics.

Reply:
I double-check in a moment, but from what I understand what you're saying is correct,

Reply:
Yes, you're correct, I've changed it to what you suggested.